src/pip/_internal/cli/cmdoptions.py

```diff
@@ -99,6 +101,122 @@ def check_dist_restriction(options: Values, check_target: bool = False) -> None:
             )
 
 
+def validate_platform_options(platform_options: List[str]) -> None:
+    """
+    Determine if platform options follow standard structures provided
+    in PEPs 425, 513, 571, 599, and 600
+
+    :param options: optparse options as a read-only Values object
+    """
+
+    def is_macos_arch(suffix: str) -> bool:
+        osx_re = re.compile(r"(?P<major>\d+)_(?P<minor>\d+)_(?P<arch>.+)")
+        match = osx_re.fullmatch(suffix)
+        if not match:
+            return False
+
+        major, minor, arch = (
+            int(match.group("major")),
+            int(match.group("minor")),
+            match.group("arch"),
+        )
+        return (major, minor) >= (10, 0) or arch in {"arm64", "intel", "x86_64"}
+
+    def is_linux_arch(prefix: str, suffix: str) -> bool:
+        """
+        determine if a platform suffix has proper content and structure
+        for details about
+        """
+        # manylinux1, manylinux2010, manylinux2014, and linux should be followed
+        # by _<architecture>
+        linux_re = re.compile(r"(?P<arch>.+)")
+        match = linux_re.fullmatch(suffix)
+        if not match:
+            return False
+
+        arch = match.group("arch")
+        if prefix == "linux":
+            return arch in {"i386", "x86_64"}
+        if prefix in {"manylinux1", "manylinux2010"}:
+            return arch in {"i686", "x86_64"}
+        if prefix == "manylinux2014":
+            return arch in {
+                "aarch64",
+                "armv71",
+                "i686",
+                "ppc64",
+                "ppc64le",
+                "s390x",
+                "x86_64",
+            }
```

Comment:
For consistency.

```suggestion
        if prefix == "linux":
            return arch in {"i386", "x86_64"}
        elif prefix in {"manylinux1", "manylinux2010"}:
            return arch in {"i686", "x86_64"}
        elif prefix == "manylinux2014":
            return arch in {
                "aarch64",
                "armv71",
                "i686",
                "ppc64",
                "ppc64le",
                "s390x",
                "x86_64",
            }
```

Reply:
https://beta.ruff.rs/docs/rules/superfluous-else-return I disliked this rule when I first saw it but I have gotten used to it.